src/lib/stores/walletStore.js

```diff
@@ -95,19 +133,42 @@ function createWalletStore() {
          */
         sign: async ({ transactionXDR, network, pincode }) => {
             try {
-                const keyManager = setupKeyManager()
-                let signedTransaction = await keyManager.signTransaction({
+                const keyManager = setupKeyManager();
+                const { keyId, publicKey } = get(walletStore);
+                
+                if (keyId === publicKey) {
+                    const kit = new StellarWalletsKit({
+                        network: WalletNetwork.TESTNET,
+                        selectedWalletId: XBULL_ID,
+                        modules: allowAllModules(),
+                    });
+                    const { address } = await kit.getAddress();
+                    
+                    // Sign the transaction using the wallet address
+                    const { signedTxXdr } = await kit.signTransaction(transactionXDR, {
+                        address,
+                        networkPassphrase: WalletNetwork.PUBLIC, // or use your specific network passphrase
```

Comment:
We're passing the network passphrase to our `walletStore`'s `sign` function as the `network` parameter. Let's use that here.

```suggestion
                        networkPassphrase: network, // or use your specific network passphrase
```